components/modals/NewTaskModal.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import TaskForm from '../forms/TaskForm';
+import { NotificationDateTimeType, TaskFormData, TaskPostData } from '../../types/task';
+import tasksApi from '../../api/tasksApi';
+import {
+  DEFAULT_NOTIFICATION_DAYS_OFFSET,
+  DEFAULT_NOTIFICATION_TIME,
+  getNotificationDateTime,
+} from '../../utils/task/taskUtils';
+import { Nullable } from '../../types/utils';
+import { Modal } from 'antd';
+
+interface Props {
+  applicationId: number;
+  setIsAddingNewTask: React.Dispatch<React.SetStateAction<boolean>>;
+  setShouldFetchData: Dispatch<SetStateAction<boolean>>;
+}
+
+function NewTaskModal({ applicationId, setIsAddingNewTask, setShouldFetchData }: Props) {
+  const [shouldDisableSaveButton, setShouldDisableSaveButton] = useState(true);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [taskFormData, setTaskFormData] = useState<Nullable<TaskFormData>>(null);
+  const [initialValues] = useState<TaskFormData>({
+    notificationTime: DEFAULT_NOTIFICATION_TIME,
+    notificationDaysOffset: DEFAULT_NOTIFICATION_DAYS_OFFSET,
+    notificationDateTimeType: NotificationDateTimeType.NONE,
+    isDone: false,
+  });
+
+  useEffect(() => {
+    if (taskFormData !== null) {
+      submit(taskFormData);
+    }
+  }, [taskFormData]);
+
+  const onCancel = () => {
+    setIsAddingNewTask(false);
+  };
+
+  const onSubmit = () => {
+    setIsSubmitting(true);
+  };
+
+  const submit = (values: TaskFormData) => {
+    const notificationDateTime = getNotificationDateTime(values);
+
+    // Note: notificationDateTime can be null
+    const taskPostData: TaskPostData = {
+      title: values.title ?? 'Task',
+      dueDate: values.dueDate?.toISOString() ?? new Date().toISOString(),
+      notificationDateTime: notificationDateTime === undefined ? new Date().toISOString() : notificationDateTime,
```

Comment:
```suggestion
      notificationDateTime: notificationDateTime === undefined ? null : notificationDateTime,
```

Should this be null instead according to you comment?

Reply:
It can be anything because the undefined case shouldn't happen (by right should throw error), I can change to null if that's clearer.